### PR TITLE
reduce caching time for the data transformation endpoints

### PR DESCRIPTION
**Pull request by Baalmart**
# reduce caching time

**_WHAT DOES THIS PR DO?_**
Reduces the caching time for the data transformation endpoints.

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
N/A

**_WHAT IS THE LINK TO THE PR BRANCH_**
https://github.com/airqo-platform/AirQo-api/tree/hf-reduce-caching-time

**_HOW DO I TEST OUT THIS PR?_**
`cd AirQo-api/src/data-mgt/node`
`npm install`
`npm run stage-mac `for Linux OR `npm run stage-mac` for Windows

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
The endpoints for data management [here](https://docs.airqo.net/airqo-platform-api/data-management).
Alternatively, you can just test using one endpoint: http://localhost:3000/api/v1/data/feeds/transform/recent?channel=689750
To confirm that things are working well, please just make a request and a subsequent one immediately, the `isCache` field will return `true` in the response.
Wait for 30 seconds (using any timer of preference) and then check out the response again, the `isCache` field should change to `false`.

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
N/A

**_ARE THERE ANY RELATED PRs?_**
N/A

### CODE REVIEW

**Comment by danielogen** [comment]
Hi @Baalmar, this looks good to me. Just some quick thoughts, is it possible to abstract the value to like “constant.js” or “.env”? So that we don't have to do a lot of changes. Thanks!

**Comment by Baalmart** [comment]
> Hi @Baalmar, this looks good to me. Just some quick thoughts, is it possible to abstract the value to like “constant.js” or “.env”? So that we don't have to do a lot of changes. Thanks!

Awesome suggestion, I am on top of it. Thanks Daniel

File: src/data-mgt/node/controllers/transform.js [hunk 1]
```diff
@@ -103,7 +103,7 @@ const data = {
                   cacheID,
                   JSON.stringify({ isCache: true, ...responseData })
                 );
-                redis.expire(cacheID, 86400);
+                redis.expire(cacheID, 30);
```
Comment (danielogen) [inline src/data-mgt/node/controllers/transform.js:106]: could this be a typo? is it 30 or 300??

File: src/data-mgt/node/controllers/transform.js [hunk 2]
```diff
@@ -148,7 +148,7 @@ const data = {
                   cacheID,
                   JSON.stringify({ isCache: true, ...responseJSON })
                 );
-                redis.expire(cacheID, 86400);
+                redis.expire(cacheID, 3600);
```
Comment (danielogen) [inline src/data-mgt/node/controllers/transform.js:151]: Also here! 3600 or 300 or 30??. just being curious!

Comment (Baalmart) [reply src/data-mgt/node/controllers/transform.js]: these are seconds. So 3600 refers to one hour. 300 is 5 minutes, 30 is 30 seconds. :)

Comment (Baalmart) [reply src/data-mgt/node/controllers/transform.js]: these are seconds. So 3600 refers to one hour. 300 is 5 minutes, 30 is 30 seconds. :)

File: src/data-mgt/node/controllers/transform.js (hunk 1 above)
Comment (Baalmart) [reply src/data-mgt/node/controllers/transform.js]: This was intentional, different endpoints have different cache expiry times...

File: src/data-mgt/node/controllers/transform.js (hunk 2 above)
Comment (danielogen) [reply src/data-mgt/node/controllers/transform.js]: ah okay... so those values are the correct ones! got it thanks @Baalmart

**Review by danielogen — APPROVED** [review]
LGTM

File: src/data-mgt/node/controllers/transform.js (hunk 1 above)
Comment (danielogen) [reply src/data-mgt/node/controllers/transform.js]: Okay, got it. thanks!

**Comment by NoahNsimbe** [comment]
Some guidance on how to test... @Baalmart @danielogen

**Comment by NoahNsimbe** [comment]
Some guidance on how to test... @Baalmart @danielogen

**Comment by Baalmart** [comment]
> Some guidance on how to test... @Baalmart @danielogen

When you run any of the endpoints from the **data-mgt** like this one for [get last entry (descriptive)](https://docs.airqo.net/airqo-platform-api/data-management#get-last-entry-descriptive), you observe the `isCache `attribute's value. The` true `value should only last for 30 seconds due to a recent change.

**Review by ja-odur — APPROVED** [review]
LGTM

**Review by sserurich — APPROVED** [review]
LGTM

**Pull request merged**